common/util/thread_pool.h

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017-2023 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_COMMON_UTIL_THREAD_POOL_H
+#define VERIBLE_COMMON_UTIL_THREAD_POOL_H
+
+#include <deque>
+#include <functional>
+#include <future>
+#include <thread>
+#include <vector>
+
+namespace verible {
+// Simple thread-pool.
+// Passing in functions, returning futures.
+//
+// Why not use std::async() ? That standard is so generic and vaguely
+// specified that in practice there is no implementation of a policy that
+// provides a thread-pool behavior with a guaranteed upper bound of cores used
+// on all platforms.
+class ThreadPool {
+ public:
+  // Create thread pool with "thread_count" threads.
+  // If that count is zero, functions will be executed synchronously.
+  ThreadPool(int thread_count) {
```

Comment:
since this is not a template class, maybe implement the ctor and dtor in the .cc file?  (These are not likely to be inline-critical)